src/towncrier/_writer.py

```diff
@@ -11,33 +11,55 @@
 
 
 def append_to_newsfile(
-    directory, filename, start_string, top_line, content, single_file=True
+    directory, filename, start_string, top_line, content, single_file
 ):
+    """
+    Write *content* to *directory*/*filename* behind *start_string*.
 
+    Double-check *top_line* (i.e. the release header) is not already in the
+    file.
+
+    if *single_file* is True, add it to an existing file, otherwise create a
+    fresh one.
+    """
     news_file = os.path.join(directory, filename)
 
-    if single_file:
-        if not os.path.exists(news_file):
-            existing_content = ""
-        else:
-            with open(news_file, encoding="utf8") as f:
-                existing_content = f.read()
-        existing_content = existing_content.split(start_string, 1)
-    else:
-        existing_content = [""]
-
-    if top_line and top_line in existing_content[-1]:
+    header, prev_body = _figure_out_existing_content(
+        news_file, start_string, single_file
+    )
+
+    if top_line and top_line in prev_body:
         raise ValueError("It seems you've already produced newsfiles for this version?")
 
-    with open(os.path.join(directory, filename), "wb") as f:
+    with open(news_file, "w", encoding="utf8") as f:
```

Comment:
I guess that this might cause the failure on windows.
I prefer to always keep the file in binary mode so that Python will not mess with newlines.

Reply:
doesn't seem to help 🤔 https://github.com/twisted/towncrier/pull/419/commits/ee54060b04fc99f525cbc10b9b8ebcfa1d4a54c3

I'll to keep digging, this makes no sense.

Reply:
ok this is funny, because now a different test is failing – one that _relied_ on newline magic: https://github.com/twisted/towncrier/runs/8167043027?check_suite_focus=true

this is gonna be a long morning